### PR TITLE
storage: GCS + KMS integration samples

### PR DESCRIPTION
**Pull request by kurtisvg**

### CODE REVIEW

File: storage/objects/main.go [hunk 1]
```diff
@@ -253,6 +253,24 @@ func writeEncryptedObject(client *storage.Client, bucket, object string, secretK
 	return nil
 }
 
+func writeWithKMSKey(client *storage.Client, bucket, object string, keyName string) error {
```
Comment (tbpg) [inline storage/objects/main.go]: Would you mind adding function comments for this and `writeEncryptedObject`? At first glance they seem the same (they aren't, but it's easy to miss how the encryption differs).

File: storage/buckets/main.go [hunk 2]
```diff
@@ -238,3 +238,17 @@ func checkRequesterPays(c *storage.Client, bucketName string) error {
 	// [END get_requester_pays_status]
 	return nil
 }
+
+func enableDefaultKMSkey(c *storage.Client, bucketName string, keyName string) error {
```
Comment (tbpg) [inline storage/buckets/main.go]: Maybe `setDefaultKMSKey`?

File: storage/buckets/main_test.go [hunk 3]
```diff
@@ -17,6 +17,7 @@ import (
 var (
 	storageClient *storage.Client
 	bucketName    string
+	kmsKeyName    string
```
Comment (tbpg) [inline storage/buckets/main_test.go]: We'll talk about testing later, but `kmsKeyName` is only used in one location. So, it's OK to define it there.

File: storage/buckets/main_test.go [hunk 4]
```diff
@@ -104,3 +106,10 @@ func TestDelete(t *testing.T) {
 		t.Fatalf("failed to delete bucket (%q): %v", bucketName, err)
 	}
 }
+
+func TestKMS(t *testing.T) {
+	testutil.SystemTest(t)
+	if err := enableDefaultKMSkey(storageClient, bucketName, kmsKeyName); err != nil {
+		t.Fatalf("failed to enable default kms key (%q): %v", bucketName, err)
+	}
```
Comment (tbpg) [inline storage/buckets/main_test.go]: Should we test writing and reading something to make sure encryption was enabled?

File: storage/buckets/main_test.go [hunk 5]
```diff
@@ -11,7 +11,9 @@ import (
 	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
 
 	"cloud.google.com/go/storage"
+	"fmt"
 	"golang.org/x/net/context"
+	"os"
```
Comment (broady) [inline storage/buckets/main_test.go]: os and fmt should go up with testing and time (standard library imports)

File: storage/objects/main_test.go [hunk 6]
```diff
@@ -16,7 +16,9 @@ import (
 	"cloud.google.com/go/storage"
 	"golang.org/x/net/context"
 
+	"fmt"
 	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	"os"
```
Comment (broady) [inline storage/objects/main_test.go]: same here

File: storage/objects/main_test.go [hunk 7]
```diff
@@ -132,9 +134,16 @@ func TestObjects(t *testing.T) {
 	key := []byte("my-secret-AES-256-encryption-key")
 	newKey := []byte("My-secret-AES-256-encryption-key")
 
+	keyRingID := os.Getenv("GOLANG_SAMPLES_KMS_KEYRING")
+	cryptoKeyID := os.Getenv("GOLANG_SAMPLES_KMS_CRYPTOKEY")
+	kmsKeyName := fmt.Sprintf("projects/%s/locations/%s/keyRings/%s/cryptoKeys/%s", tc.ProjectID, "global", keyRingID, cryptoKeyID)
```
Comment (broady) [inline storage/objects/main_test.go]: need to skip this if these env vars don't exist. maybe belongs in a different (skippable) test?

See https://golang.org/pkg/testing/#T.Skip

**Comment by broady** [comment]
LGTM, looking into permissions issue now.

**Comment by frankyn** [comment]
Hey @broady, any luck with the permission issue?

**Comment by tbpg** [comment]
All commit authors are Googlers. So, I added the `cla: yes` label.

I added a commit to move the KMS test before the delete test since it was causing the KMS test to fail. Will let Kokoro run again.

**Comment by frankyn** [comment]
thanks @tbpg, based on the error message: "Error 403: Permission denied on Cloud KMS key, forbidden", I think the project owning the kms key being used for tests hasn't granted the [GCS managed service account](https://developers.google.com/apis-explorer/#p/storage/v1/storage.projects.serviceAccount.get) the `roles/cloudkms.cryptoKeyEncrypterDecrypter` role. Each project has a GCS managed service account.

**Comment by frankyn** [comment]
Follow-up comment from GVC conversation with @tbpg about related documentation. 
https://cloud.google.com/storage/docs/encryption/customer-managed-keys

**Comment by tbpg** [comment]
Thanks! I added the permission for all of the test projects.

```
for i in $(seq 2 10); do
gcloud projects add-iam-policy-binding golang-samples-tests-$i --member serviceAccount:golang-samples-tests-$i@gs-project-accounts.iam.gserviceaccount.com --role roles/cloudkms.cryptoKeyEncrypterDecrypter
done
```

I rebased, pushed, and will wait for Kokoro again. It will probably fail due to DLP quota, though :disappointed: (working on it).

**Comment by frankyn** [comment]
Thanks @tbpg!

**Comment by tbpg** [comment]
:smile: Test failures caused by DLP. Merging.

**Pull request merged**